core/http/src/uri/uri.rs

```diff
@@ -33,6 +33,14 @@ use crate::uri::error::{Error, TryFromUriError};
 /// methods of the internal structure.
 ///
 /// [RFC 7230]: https://tools.ietf.org/html/rfc7230
+///
```

Comment:
I think it's weird that it implements one but not the other. I can't think of any utility this offers. I would suggest either implementing both (using parse_any for deserialize) or neither.